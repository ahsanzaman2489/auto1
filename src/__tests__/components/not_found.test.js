import React from 'react';
import NotFoundComponent from '../../components/404';
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from 'react-router-dom';


const setup = (props = {}, state = null) => {
    const defaultProps = {};
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<NotFoundComponent {...setUpProps}/>);
    const tree = () => renderer.create(<Router><NotFoundComponent {...setUpProps}/></Router>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree}
};

describe("Not found component", () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });

    it('Should render not found component', () => {
        const {wrapper} = setup();
        expect(wrapper.find('.pagenotfound').length).toBe(1);
        expect(wrapper.find('.logo').length).toBe(1);
        expect(wrapper.find('.content404').length).toBe(1);
        expect(wrapper.find('NavLink').length).toBe(1);
        expect(wrapper.find('.content404').childAt(0).text()).toBe('404 - Not Found')
        expect(wrapper.find('h1').length).toBe(1);
        expect(wrapper.find('h1').length).toBe(1);
    });

});

