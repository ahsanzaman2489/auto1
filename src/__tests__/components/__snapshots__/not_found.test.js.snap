// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found component should match with snapshot 1`] = `
<div
  className="container"
>
  <div
    className="pagenotfound"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="logo.png"
      />
    </div>
    <div
      className="content404"
    >
      <h1>
        404 - Not Found
      </h1>
      <p>
        Sorry, the page you are looking for does not exist. You can always go back to the 
        <a
          aria-current={null}
          className={undefined}
          href="/cars/list/"
          onClick={[Function]}
          style={undefined}
        >
          homepage
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
